### PR TITLE
Shell script wrappers

### LINKED ISSUE #565: Generating shell scripts instead of functions for wrappers without specifying on the container.yaml
As the wrapper_scripts option is now enabled by default, shpc will only create scripts in cases the container.yaml file has the docker_scripts/singularity_scripts tag.

My suggestion would be that, instead of creating functions for calling the containers, simple default shell scripts with such content be created instead. This way, `slurm` and others will be happy to call it. 

The current default is to only generate shell scripts if there's enough info on the container.yaml.

Changes needed:
- The module would need to export a path, which is not done now.
- Each function (like container_run, container_exec, container_container, container_shell) gets created as a shell script instead of a function.

What do you think? That's better than to have to add a script/wrapper entry to every single container in the registry.

### PR DESCRIPTION
**Pull request by vsoch**
This will close #565

### CODE REVIEW

**Comment by muffato** [comment]
Hi. I'm facing an issue when testing this. For a certain software, I've defined a custom wrapper script for one of its aliases. Very much like `vanessa/salad` except that it doesn't work :/
Somehow, the file that is being created has a single line, which contains the full path (with no line return) of the template it was supposed to load and substitute. Any idea what could be causing that ? I can't trivially share it because it's not an open-source software, and one has to sign the licence agreement and build their own container.

**Comment by muffato** [comment]
Oh, the issue is also on the main branch ! it must come from one of the recent PRs, then

**Comment by vsoch** [comment]
Can you give me the command to reproduce?

**Comment by vsoch** [comment]
I have a feeling a script name is being provided as the variable instead of the template somewhere. If I can reproduce I can get it into this PR and fix asap!

**Comment by muffato** [comment]
I could clearly reproduce the issue on `vanessa/salad`, cf #588

**Comment by vsoch** [comment]
Taking a look now.

File: shpc/main/modules/templates/docker.tcl [hunk 1]
```diff
@@ -101,6 +97,12 @@ set-alias {|module_name|}-shell "${shellCmd}"
 {% endfor %}
 }{% endif %}
 
+{% if wrapper_scripts %}{% else %}
+set-alias {|module_name|}-container "echo ${containerPath}"
+
+# set_shell_function takes bashStr and cshStr
```
Comment (muffato) [inline shpc/main/modules/templates/docker.tcl:103]:
```suggestion
```

**Review by muffato — APPROVED** [review]
`set_shell_function` is a Lua thing, right ? I can't see any reference to it in the rest of `docker.tcl` and `singularity.tcl`

File: shpc/main/modules/templates/singularity.tcl [hunk 2]
```diff
@@ -108,7 +103,11 @@ set-alias {|module_name|}-shell "${shellCmd}"
 {% endfor %}
 }{% endif %}
 
-# A customizable exec function
+{% if wrapper_scripts %}{% else %}# set_shell_function takes bashStr and cshStr
```
[the remark this answers is not in the document]
Comment (vsoch) [reply shpc/main/modules/templates/singularity.tcl:106]: Good catch! That likely was an old comment copy pasted from the other file.

**Pull request merged**